src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -244,12 +264,13 @@ internal enum NodeFlags : byte
             ContainsSkippedText = 1 << 3,
             ContainsAnnotations = 1 << 4,
```

Comment:
A non-inherited form of `ContainsAnnotations` would be an excellent candidate for another flag, since it would allow us to avoid a large number of costly operations in `GetAnnotations` for cases where we are searching for annotations in deeply nested nodes.

Reply:
@sharwell Exploring that idea in: https://github.com/dotnet/roslyn/pull/72104